### ISSUE TITLE
Version with Python support?

### ISSUE THREAD

**Issue by 501st-alpha1** [dcycle/docker-ledger#9]
Would it be possible to get a version of this image with Python support?  Or would that depend on getting a version with Python support built as an Alpine package?

I assume not everyone would want to include Python support, so I'd be content with using a separate tag for that.

**Comment by alberto56** [comment]
Hi @501st-alpha1 I'm not sure it's a good idea to include Python support in this image. The Docker approach is generally to use different images/containers if you need different software, something like. Unless I'm missing something, I would suggest you [use the official python image](https://hub.docker.com/_/python), something like this:

Let's say you have a Python script called `your-script.py` in your folder which generates a file called sample.dat, you could something like this:

    docker run -it --rm --name my-running-script \
      -v "$PWD":/usr/src/myapp -w /usr/src/myapp \
      python:3 python your-script.py
    docker run --rm -v $PWD:/data \
      dcycle/ledger:1 -f /data/sample.dat reg

If you still think you need python in this image, please provide a detailed usecase which cannot be met by using the official python docker image.

**Comment by 501st-alpha1** [comment]
Sorry, I wasn't clear.  I meant could we get an image that has Ledger built with Python support.  [Ledger supports embedding Python within your Ledger file.](https://www.ledger-cli.org/3.0/doc/ledger3.html#Embedded-Python)

Sorry for the confusion.

**Comment by alberto56** [comment]
Thanks for the clarification! Yes that would definitely be a good addition, I added a new branch dev/9 and [this pull request](https://github.com/dcycle/docker-ledger/pull/10/files). [The automated tests are passing](https://app.circleci.com/pipelines/github/dcycle/docker-ledger/21/workflows/6f6c615b-14c0-4032-96a7-871ae34ac052/jobs/22) so can you please test this and see if it works?

**Comment by alberto56** [comment]
Also, I normally like to add some examples and run automated tests on them, at https://github.com/dcycle/docker-ledger/tree/master/examples

Perhaps you can suggest a very simple example which uses python, which we can add to the examples, and the automated tests to make sure this works.

**Comment by 501st-alpha1** [comment]
Sure, I can take a look at that when I get a chance, and can see about adding some examples.

However, I don't think the PR will work as is, because Ledger also has to be built with Python support.  I think it's `./acprep --python` or something if building from source; I don't think the current Alpine package was built with that enabled.

**Comment by alberto56** [comment]
I [updated the pull request](https://github.com/dcycle/docker-ledger/pull/10/files) with a simple test that should pass, and like you said, it fails with:

```
While parsing file "/data/python.dat", line 4:
Error: 'python' directive seen, but Python support is missing
While parsing file "/data/python.dat", line 16:
While evaluating value expression:
  check_path(value)
  ^^^^^^^^^^^^^^^^^
```

[as can be seen in the failing test report](https://app.circleci.com/pipelines/github/dcycle/docker-ledger/22/workflows/135ae1a5-920d-4af1-b49b-8e8e2ed31620/jobs/23)

I opened https://github.com/ledger/ledger/issues/1886 to get more guidance on this, as I cannot find instructions online. If you figure out how to do this, please let me know and I'll install it, seems like a great feature.

**Comment by 501st-alpha1** [comment]
Yeah I can't remember now where I learned how to build it with Python support, and I don't see any obvious instructions in the README or the docs.  But I do have it working in my locally built-from-source version of Ledger, so presumably I could figure out how to do it again.

The main question is whether you'd prefer to have a Dockerfile that builds Ledger from source, or if you want to try to get an official Alpine package built that way.  I don't have much experience with Alpine packaging, so I'm not sure what's involved in the latter, but if you want to try the former, I might be able to get that working.

**Comment by alberto56** [comment]
The former (built from source) is fine; feel free to try a pull request which looks a bit like dev/9 but where tests pass, or paste your instructions here.

**Comment by obscurerichard** [comment]
What you are looking for might look a bit like https://github.com/obscurerichard/ledger-pyreport/blob/ef2e8f4ded460839ddb45b2c5dfe6c928ba1e9ca/Dockerfile#L1-L16 @501st-alpha1 - I built Ledger from source with Python support in Docker in another project. It's puzzling that the Alpine `ledger` lacks Python support.

**Comment by alberto56** [comment]
I think the philosophy of Alpine is to be as small as possible, which is why it comes without Python.

I just updated the [dev/9 branch in a PR](https://github.com/dcycle/docker-ledger/pull/19/files) at https://github.com/dcycle/docker-ledger/pull/19, just to remember what the problem, and we can see that the tests [still do not pass](https://app.circleci.com/pipelines/github/dcycle/docker-ledger/122/workflows/15756fc0-bb5d-4a9b-8c1a-9f22864095ae/jobs/124) with the problem:

> Error: 'python' directive seen, but Python support is missing

The only reference to this error I could find on Google is [here](https://github.com/dmitry-merzlyakov/nledger/blob/master/nledger.md#enable-python-integration), but it's not clear to me how to fix this in this image.

I also tried https://stackoverflow.com/a/62555259 but I'm still getting the error "Error: 'python' directive seen, but Python support is missing".

I added the issue https://github.com/ledger/ledger/issues/2401 in the Ledger queue in case the maintainers of Ledger might have guidance on this.
